autoload/asyncomplete/sources/tags.vim

```diff
@@ -19,14 +19,18 @@ function! asyncomplete#sources#tags#completor(opt, ctx)
         return
     endif
 
+    let l:tags_tempname = tempname()
+
     let l:matches = {}
 
     for l:tag_file in l:tag_files
-        let l:lines = readfile(l:tag_file)
+        silent exec 'grep! -P "^' . l:kw . '[^\t]*\t" ' . l:tag_file . ' > ' . l:tags_tempname . ' 2>/dev/null'
```

Comment:
This would be blocking vim. Can we use async job apis instead?

Can embed [async.vim](https://github.com/prabirshrestha/async.vim) or just create another wrapper here so it works with both vim and neovim.

Reply:
I get results (i.e., the list of matching tags) in under one second, but maybe on a slower computer the block might be noticeable.

I'll check the job API and see what I can do.

Reply:
Doesn't seem to work on windows most likely coz `2>/dev/null` isn't working.

The experience isn't good in windows. Since the command is sync it hangs a for a sec, then opens a new window which can be see in the taskbar and then closes.

The best would be to use the jobs api which solves both the problems. Feel free to make it only work on mac/linux and I can modify it to work in windows with findstr. You might want to send the -1 check in different PR so it can be merged.